loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,430 @@
+* [quilt_loader](#the-quilt_loader-field) — Information related to loading the mod
+    * [group](#the-group-field) — The Maven group id
+    * [id](#the-id-field) — The mod id
+    * [provides](#the-provides-field) — Alternative mod ids provided by this mod
+    * [version](#the-version-field) — The mods version
+    * [entrypoints](#the-entrypoints-field) — Collection of entrypoints
+    * [plugins](#the-plugins-field) — Collection of plugins
+    * [jars](#the-jars-field) — Array of nested JARs to be loaded
+    * [language_adapters](#the-language_adapters-field) — Array of language adapters
+    * [depends](#the-depends-field) — Collection of mod dependencies
+    * [breaks](#the-breaks-field) — Collection of mods that this mod is incompatible with
+    * [repositories](#the-repositories-field) — Array of maven repositories
+    * [metadata](#the-metadata-field) — Extra information about this mod and/or its authors
+        * [name](#the-name-field) — Human-readable name of this mod
+        * [description](#the-description-field) — Human-readable description of this mod
+        * [contributors](#the-contributors-field) — Collection of contributors to this mod
+        * [contact](#the-contact-field) — Collection of contact information
+        * [license](#the-license-field) — One or more licenses this project is under
+        * [icon](#the-icon-field) — The icon or icons associated with this project
+* [mixin](#the-mixin-field) — Path(s) to mixin config file(s)
+* [access_widener](#the-access_widener-field) — Path(s) to accesswidener file(s)
+* [minecraft](#the-minecraft-field) - Minecraft related options
+    * [environment](#the-environment-field) — What game environments this mod applies to
+    
+## The `schema_version` field
+| Type   | Required |
+|--------|----------|
+| Number | True     |
+
+The quilt mod file schema version to be used for parsing this file. Currently, the only valid version is 1.
+
+## The `quilt_loader` field
+| Type   | Required |
+|--------|----------|
+| Object | True     |
+
+Information necessary for the mod loading process.
+
+### The `group` field
+| Type   | Required |
+|--------|----------|
+| String | True     |
+
+A unique identifier for the organization behind or developers of the mod. See Maven's [guide to naming conventions](https://maven.apache.org/guides/mini/guide-naming-conventions.html).
+
+### The `id` field
+| Type   | Required |
+|--------|----------|
+| String | True     |
+
+A unique identifier for the mod or library defined by this file, matching the `^[a-z][a-z0-9-_]{1,63}$` regular expression. Best practice is that mod ID's are in snake_case.
+
+### The `provides` field
+| Type         | Required |
+|--------------|----------|
+| Array/String | False     |
+
+Either a single identifier or an array of identifiers that this mod also contains. These could be old or alternative mod ids. They should follow [the convention established for mod ids](#the-mod_id-field).
```

Comment:
So one problem I've had when doing Quilt Loader is that our versions have diverged but our API is only compatible with Fabric Loader's API at a lower version. In the future, maybe the opposite happens and Fabric overtakes us, and then we have even more issues.

I think it should be possible to specify that you are providing a specific version of a dependency.

Reply:
I've changed the `provides` field to use a dependency object instead of simple mod ids so that developers can provide various versions under different circumstances. I also added to the example to show what this might look like.